client-app/shared/catalog/composables/useBreadcrumbs.ts

```diff
@@ -0,0 +1,25 @@
+import { IBreadcrumbsItem } from "./../types/index";
+import { Breadcrumb } from "@core/api/graphql/types";
+
+export default () => {
+  function buildBreadcrumbs(breadcrumbs: Breadcrumb[]) {
+    const result: IBreadcrumbsItem[] = [{ url: "/", title: "Home" }];
+
+    const productBreadcrumbs = breadcrumbs.map((x) => {
+      return {
+        title: x?.title,
+        url: x?.typeName === "CatalogProduct" ? "/product/" + x.itemId : "/catalog/" + x?.seoPath,
```

Comment:
Please consider moving out all logic for routing link construction into one place (specialized utility class will be a good solution)

Reply:
It is the separated utility already. Or move it to a simple helper class instead of a Composable API approach?